app/controllers/api/v3/subject_areas_controller.rb

```diff
@@ -2,7 +2,7 @@ module API
   module V3
     class SubjectAreasController < API::V3::ApplicationController
       def index
-        render jsonapi: SubjectArea.active, fields: fields_param, include: params[:include], class: CourseSerializersService.new.execute[:v3]
+        render jsonapi: SubjectArea.active.includes(subjects: [:financial_incentive]), fields: fields_param, include: params[:include], class: CourseSerializersService.new.execute[:v3]
```

Comment:
Do you think it should be the API that includes `subjects: [:financial_incentive]` or should this be done by the caller via params? 🤔

Reply:
the find client call is `/api/v3/subject_areas?include=subjects` so using that to determine the include is still not sufficient. I think it's a very hard problem to solve given our api can include associations based on what the client sends, optimising all these will be a challenge. 

For now no one is really using this endpoint but us. But later when we get more traffic I feel the much bigger problem should be revisited.

Reply:
👍